File: pbx-platform/services/frontend/types/customer.ts
export interface Customer {
    id: number;
    name: string;
    phone: string;
    email: string | null;
    company_id: number | null;
    company_name: string | null;
    group: string;
    memo: string | null;
    created_at: string;
    last_call_at: string | null;
    deactivated_at: string | null;
}

export interface CustomerGroup {
    id: string;
    label: string;
    color: string;
}

export interface CallHistory {
    id: string;
    date: string;
    direction: "inbound" | "outbound" | "internal";
    duration: string;
}

File: pbx-platform/services/api/app/schemas/customer.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel

class CustomerCreate(BaseModel):
    name: str
    phone: str
    email: Optional[str] = None
    company_id: Optional[int] = None
    group: str = "normal"
    memo: Optional[str] = None

class CustomerUpdate(BaseModel):
    name: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    company_id: Optional[int] = None
    group: Optional[str] = None
    memo: Optional[str] = None

class CustomerResponse(BaseModel):
    id: int
    name: str
    phone: str
    email: Optional[str]
    company_id: Optional[int]
    company_name: Optional[str] = None
    group: str
    memo: str
    created_at: datetime
    last_call_at: Optional[datetime]
    deactivated_at: Optional[datetime]

    model_config = {"from_attributes": True}
